### PR TITLE
feat: add Host struct and convert from String and Url

### PR DESCRIPTION
**Pull request by Frando**
Implements #15

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -97,9 +98,61 @@ mod accept {
...
+        fn from(url: &Url) -> Self {
+            Self(
+                url.host_str()
+                    .expect("URL has to include a host part.")
+                    .into(),
+            )
+        }
+    }
+}
+
```
Comment (yoshuawuyts) [inline src/lib.rs]: Rather than re-declaring the #[`cfg(feature)]` blocks repeatedly, it can be applied to blocks directly. See https://github.com/async-rs/async-std/blob/master/src/utils.rs#L83-L105 for the generated macro code we use in `async-std` where we do a similar thing.

**Comment by Frando** [comment]
Yes, I was also unsure about the feature flag. If there's no objections towards no feature flag I'd remove it.

**Comment by dignifiedquire** [comment]
thank you, I agree, no need for the feature flag

**Comment by Frando** [comment]
Ok, I removed the feature flag!

**Comment by yoshuawuyts** [comment]
ping @dignifiedquire can we merge this?

File: src/lib.rs [hunk 2]
```diff
@@ -97,9 +98,60 @@ mod accept {
...
+    pub struct Host(String);
+
+    impl Host {
+        /// The host as string. Consumes self.
+        pub fn as_string(self) -> String {
+            self.0
+        }
+    }
+
+    impl From<&str> for Host {
```
Comment (dignifiedquire) [inline src/lib.rs]: can we add an impl for AsRef<str> that would cover things like &String as well

File: tests/google.rs [hunk 3]
```diff
@@ -56,7 +56,8 @@ async fn fetch_google() {
     // Send off the request by first negotiating an SSL handshake, then writing
     // of our request, then flushing, then finally read off the response.
     let connector = async_native_tls::TlsConnector::new();
-    let mut socket = t!(connector.connect("google.com", socket).await);
+    let url = url::Url::parse("https://google.com/").unwrap();
```
Comment (dignifiedquire) [inline tests/google.rs]: why this change? is it actually required?

File: src/lib.rs (hunk 2 above)
Comment (Frando) [reply src/lib.rs]: I added an `impl From<&String> for Host`. Or did you mean using AsRef directly? Couldn't get that to work with From (for size not known at compile time), but also can't think of cases where it would be needed?

File: tests/google.rs (hunk 3 above)
Comment (Frando) [reply tests/google.rs]: Ah, I added this for testing that things worked with passing an Url struct. It's not required. If you want I remove it.

File: src/lib.rs (hunk 2 above)
Comment (dignifiedquire) [reply src/lib.rs]:
```rust
    impl<T: AsRef<str>> From<T> for Host {
        fn from(host: T) -> Self {
            Host(host.as_ref().to_string())
        }
    }
```

this would work, but because `Url` implements `AsRef<str>` as well, it ends up conflicting. So leaving it as is for now.

**Comment by dignifiedquire** [comment]
Thank you, released as `0.3.3`
